src/main/java/com/flightstats/hub/app/ShutdownManager.java

```diff
@@ -38,25 +46,25 @@ public boolean shutdown(boolean useLock) throws Exception {
         statsdReporter.event("Hub Restart Shutdown", "shutting down", tags);
         statsdReporter.mute();
 
-        HubHealthCheck healthCheck = HubProvider.getInstance(HubHealthCheck.class);
-        if (healthCheck.isShuttingDown()) {
+        if (this.hubHealthCheck.isShuttingDown()) {
```

Comment:
Not sure of that. Is this convention documented somewhere? If there is a local variable/argument by the same name, instance variable will be ignored. "this" keyword ensures that use of instance variables.

Reply:
You're right that local variables with the same name will cause the instance variables to be shadowed, but I think that kind of duplicate naming is confusing and should be avoided where possible. The only exception to that case is in constructors where the main (and hopefully only) purpose is to map arguments/locals to the instance variables. I usually opt to use clear, distinct names and drop `this` as I find it makes the code a little noisy and less readable.

We don't have this stylistic choice documented anywhere; it's been a common convention on the trips teams and seems to generally be the case in the hub, iirc. This is another decision that we should get consensus on so that code we touch is consistent. @Paul-Hess and @dkleckler?

Reply:
The only place using ```this``` tends to be important is in the constructor because of shadowing. The constructor is about the only place where a shadowing variable of the same name is even reasonable. Anywhere else, there shouldn't be any reason to have a local variable of the same name as an instance variable (and if you did, just pick a different name). 

Seeing an extraneous ```this``` then makes me think the clarification exists because of a conflict and I end up confusedly looking for where the conflicting variable is defined.

IntelliJ does a good job of colorization so there's little chance of confusion when you're working in the code.

Reply:
I did a quick sweep on hub code and git rid of redundant this and final keywords to maintain consistency across the app.